packages/bot-skeleton/src/services/tradeEngine/trade/Purchase.js

```diff
@@ -42,37 +38,53 @@ export default Engine =>
                     buy_price: buy.buy_price,
                 });
             };
-            const action = () => api_base.api.send({ buy: id, price: askPrice });
-            this.isSold = false;
-            contractStatus({
-                id: 'contract.purchase_sent',
-                data: askPrice,
-            });
 
-            if (!this.options.timeMachineEnabled) {
-                return doUntilDone(action).then(onSuccess);
+            function hasBlockOfType(targetType, workspace) {
+                const allBlocks = workspace.getAllBlocks();
+                return allBlocks.some(block => block.type === targetType);
+            }
+
+            const workspace = Blockly.getMainWorkspace();
```

Comment:
@maryia-matskevich-deriv i think `getMainWorkspace()` won't give the reference for workspaces on bot..
For `tab 1 `- `dashboard tab` we need to use `Blockly.mainWorkspace`
For `tab 2` -  `bot-builder tab` we need to use `Blockly.derivWorkspace`

Reply:
thank you🙂

Reply:
has changed to `Blockly.derivWorkspace`✅